srml/babe/src/lib.rs

```diff
@@ -162,6 +169,59 @@ decl_storage! {
 		/// epoch.
 		SegmentIndex build(|_| 0): u32;
 		UnderConstruction: map u32 => Vec<[u8; 32 /* VRF_OUTPUT_LENGTH */]>;
+
+		/// Temporary value (cleared at block finalization) which is true
+		/// if per-block initialization has already been called for current block.
+		Initialized get(initialized): Option<bool>;
+	}
+	add_extra_genesis {
+		config(authorities): Vec<(AuthorityId, BabeWeight)>;
+		build(|
+			storage: &mut sr_primitives::StorageOverlay,
+			_: &mut sr_primitives::ChildrenStorageOverlay,
+			config: &GenesisConfig,
+		| {
+			use primitives::storage::well_known_keys;
+			use sr_primitives::traits::TypedKey;
+
+			runtime_io::with_storage(storage, || {
+				let mut authorities = config.authorities.clone();
+
+				// if session module is here AND it has mutated initial validators list, then we want to sync
+				// it with our authorities list
+				let session_validators_keys: Vec<(T::AccountId, T::Keys)> = srml_support::storage::unhashed::get_raw(
+					well_known_keys::MUTATED_SESSION_VALIDATORS_KEYS,
+				).and_then(|v| Decode::decode(&mut &v[..]))
+					.expect("unable to decode initial session validators keys");
```

Comment:
This expect is incorrect && will panic if babe is used without sessions module or if validators aren't updated